tests/test_utils.py

```
@@ -608,3 +610,40 @@ def test_code_rabbit() -> None:
+    # n_trials x time shape
+    arr = np.array([[1, 2, 3], [4, 5, 6]])
+    result = arr.flatten()
+    assert np.array_equal(result, np.array([1, 2, 3, 4, 5, 6]))
+
+
+def test_get_tau() -> None:
+    import matplotlib.pyplot as plt
+
+    x = np.arange(1, 100)
+    tau = 5
+    data_with_noise = exponential_decay(x, tau=tau) + np.random.normal(0, 0.1, size=99)
```

Ruff (0.14.4)

Undefined name `exponential_decay`

(F821)